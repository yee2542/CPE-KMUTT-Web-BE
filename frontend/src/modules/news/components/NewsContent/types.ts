import { CardVariant } from 'common/components/Card/types';

export interface NewsContentProps {
  id?: string;
  tumbnail?: string;
  content?: string;
  title?: string;
  postDate?: string;
  file?: {
    title?: string;
    link?: string;
  }[];
  contents?: JSX.Element[];
  connections?: CardNewsConnectionProps[];
}

export interface CardNewsConnectionProps {
  id: string;
  title?: string;
  description?: string;
  date?: string;
  links?: string;
  variant?: CardVariant;
}
