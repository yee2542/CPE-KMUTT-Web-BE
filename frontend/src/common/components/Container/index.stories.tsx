import { Meta } from '@storybook/react/types-6-0';
import React from 'react';
import Container from '.';

const meta: Meta = {
  title: 'Commmon/Container',
  component: Container,
};

export default meta;

export const Default: React.FC = () => <Container>Container</Container>;
