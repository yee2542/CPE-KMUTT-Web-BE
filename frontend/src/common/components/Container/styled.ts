import BREAKPOINTS from 'common/constants/breakpoints';
import styled from 'styled-components';

export const Containers = styled.div`
  display: flex;
  width: 100%;
  max-width: ${BREAKPOINTS.IPAD_LANDSCAPE};
  margin: 0 auto;
  flex-wrap: wrap;
`;
