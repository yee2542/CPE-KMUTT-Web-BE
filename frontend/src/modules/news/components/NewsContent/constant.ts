import { CardVariant } from 'common/components/Card/types';

import { NewsContentProps, CardNewsConnectionProps } from './types';

export const constants: CardNewsConnectionProps[] = [
  {
    id: '1',
    title: 'การรับเข้าศึกษา',
    description:
      'ประกาศรายชื่อผู้มีสิทธิ์เข้าศึกษา โครงการ Active Recruitment รอบที่ 2 ประจำปีการศึกษา 2563',
    date: '28 ตุลาคม 2563',
    thumbnail: '/images/thumbnail.png',
    variant: CardVariant.primary,
  },
  {
    id: '2',
    title: 'การรับเข้าศึกษา',
    description:
      'ประกาศรายชื่อผู้มีสิทธิ์เข้าศึกษา โครงการ Active Recruitment รอบที่ 2 ประจำปีการศึกษา 2563',
    date: '28 ตุลาคม 2563',
    variant: CardVariant.normal,
  },
  {
    id: '3',
    title: 'การรับเข้าศึกษา',
    description:
      'ประกาศรายชื่อผู้มีสิทธิ์เข้าศึกษา โครงการ Active Recruitment รอบที่ 2 ประจำปีการศึกษา 2563',
    date: '28 ตุลาคม 2563',
    variant: CardVariant.normal,
  },
];

export const NEWS_CONTENT: NewsContentProps = {
  id: '1',
  title: 'ประกาศรายชื่อผู้มีสิทธิ์สอบสัมภาษณ์ โครงการ Active Recruitment รอบที่ 2',
  content:
    'ภาควิชาวิศวกรรมคอมพิวเตอร์ คณะวิศวกรรมศาสตร์มหาวิทยาลัยเทคโนโลยีพระจอมเกล้าธนบุรี เริ่มก่อตั้งและเปิดสอนในระดับปริญญาตรีตั้งแต่ปีการศึกษา 2531แต่ได้รับการอนุมัติให้จัดตั้งหน่วยงาน ภาควิชาวิศวกรรมคอมพิวเตอร์และประกาศในพระราชกฤษฎีกา เมื่อวันที่ 11 กุมภาพันธ์ 2540ภาควิชาวิศวกรรมคอมพิวเตอร์ เปิดสอนทั้งในระดับปริญญาตรี ปริญญาโท และปริญญาเอกโดยในระดับปริญญาตรี แบ่งเป็น 2 หลักสูตร คือ หลักสูตรวิศวกรรมศาสตรบัณฑิตสาขาวิชาวิศวกรรมคอมพิวเตอร์(ปกติ) และหลักสูตรวิศวกรรมศาสตรบัณฑิตสาขาวิศวกรรมคอมพิวเตอร์ (หลักสูตรนานาชาติ) ส่วนในระดับปริญญาโท คือหลักสูตรวิศวกรรมศาสตรมหาบัณฑิต และวิทยาศาสตรมหาบัณฑิตสาขาวิชาวิศวกรรมคอมพิวเตอร์ และในระดับปริญญาเอก คือ หลักสูตรปรัชญาดุษฎีบัณฑิตสาขาวิชาวิศวกรรมไฟฟ้าและคอมพิวเตอร์',
  postDate: '12 สิงหาคม 2563',
  tumbnail: '/images/thumbnail.png',
  file: [
    {
      title: 'CPE2021_AR_0.pdf',
      link: '/',
    },
    {
      title: 'CPE2021_AR_0.pdf',
      link: '/',
    },
  ],
};
