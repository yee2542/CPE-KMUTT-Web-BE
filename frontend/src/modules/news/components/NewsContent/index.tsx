import React from 'react';

import Link from 'next/link';
import { useRouter } from 'next/router';
import { HiOutlineArrowRight } from 'react-icons/hi';
import { VscFilePdf } from 'react-icons/vsc';

import Button from 'common/components/Button';
import Card from 'common/components/Card';
import Container from 'common/components/Container';
import CardsLayout from 'common/components/Layouts/CardsLayout';
import {
  STATIC_NEWS_ID_LINK,
  STATIC_NEWS_ID_LINK_PATTERN,
  STATIC_NEWS_LINK,
} from 'common/constants/links';

import {
  NewsContentTumbnail,
  NewsContentFileContainer,
  NewsContentFileHeader,
  NewsContentFileList,
  NewsContentFileName,
  NewsContentFileNameContainer,
  NewsContentHeader,
  NewsContentPostDate,
  NewsContentStyle,
  OtherNewsContainer,
  OtherNewsHeaderContainer,
  OtherNewsHeader,
} from './styled';
import { NewsContentProps } from './types';

const NewsContent: React.FC<NewsContentProps> = (props) => {
  const router = useRouter();

  function gotoAllNews() {
    router.push(STATIC_NEWS_LINK);
  }

  function onCardClick(newsId: string) {
    router.push(STATIC_NEWS_ID_LINK.replace(STATIC_NEWS_ID_LINK_PATTERN, newsId));
  }

  return (
    <NewsContentStyle>
      <Container>
        <NewsContentHeader>{props.title}</NewsContentHeader>
        <NewsContentPostDate>{props.postDate}</NewsContentPostDate>

        {/* canvas */}
        {props.tumbnail && (
          <NewsContentTumbnail
            style={{
              backgroundImage: `url(${props.tumbnail})`,
            }}
          />
        )}

        {/* dyanmic contents */}
        {props.contents}

        {/* files & download */}
        {props.file && (
          <NewsContentFileContainer>
            <NewsContentFileHeader>ไฟล์แนบ ({props.file?.length})</NewsContentFileHeader>
            {props.file?.map((data, index) => (
              <NewsContentFileList key={index}>
                <NewsContentFileNameContainer>
                  <VscFilePdf
                    style={{ marginRight: '16px' }}
                    color="#DD2025"
                    size="24px"
                  />
                  <NewsContentFileName>{data.title}</NewsContentFileName>
                </NewsContentFileNameContainer>
                <Link href={data.link ?? '/'}>ดาวน์โหลด</Link>
              </NewsContentFileList>
            ))}
          </NewsContentFileContainer>
        )}

        {/* connections */}
        {props.connections && (
          <OtherNewsContainer>
            <OtherNewsHeaderContainer>
              <OtherNewsHeader>ประกาศอื่นๆ</OtherNewsHeader>
              <Button $color="borderless" onClick={gotoAllNews}>
                แสดงทั้งหมด
                <HiOutlineArrowRight
                  className="Icon"
                  style={{ marginLeft: '15px' }}
                  size="20px"
                />
              </Button>
            </OtherNewsHeaderContainer>
            <CardsLayout>
              {props.connections.map((data) => (
                <Card
                  onClick={() => onCardClick(data.id)}
                  key={'news-connections-' + data.id}
                  title={data.title}
                  description={data.description}
                  date={data.date}
                  thumbnail={data.thumbnail}
                  id={data.id}
                  variant={data.variant}
                />
              ))}
            </CardsLayout>
          </OtherNewsContainer>
        )}
      </Container>
    </NewsContentStyle>
  );
};

export default NewsContent;
