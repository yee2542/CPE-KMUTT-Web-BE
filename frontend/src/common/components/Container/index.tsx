import React from 'react';
import { Containers } from './styled';
import { ContainerProps } from './types';

// Example components
const Container: React.FC = ({ children }: ContainerProps) => {
  return <Containers>{children}</Containers>;
};

export default Container;
