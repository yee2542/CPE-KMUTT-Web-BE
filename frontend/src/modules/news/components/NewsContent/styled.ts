import styled from 'styled-components';

import BREAKPOINTS from 'common/constants/breakpoints';
import COLORS from 'common/constants/colors';

export const NewsContentStyle = styled.div`
  display: block;
  font-family: 'Kanit', sans-serif;
  width: 100%;
  background-color: #fafafa;
  margin-top: 36px;
  @media (max-width: ${BREAKPOINTS.IPAD_LANDSCAPE}) {
    padding: 0 25px;
  }
`;

export const NewsContentHeader = styled.h1`
  font-size: 24px;
  font-weight: 700;
  color: ${COLORS.GRAY_1};
  width: 100%;
  margin-bottom: 8px;
  @media (max-width: ${BREAKPOINTS.IPAD_LANDSCAPE}) {
    font-size: 20px;
  }
`;

export const NewsContentPostDate = styled.div`
  font-size: 14px;
  font-weight: 500;
  color: ${COLORS.GRAY_2};
  margin-bottom: 48px;
`;

export const NewsContentTumbnail = styled.div`
  background-color: ${COLORS.GRAY_2};
  width: 100%;
  height: 350px;
  margin-bottom: 36px;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
`;

export const NewsContentContainer = styled.div`
  font-size: 14px;
  font-weight: 500;
  color: ${COLORS.GRAY_1};
  margin-bottom: 32px;
`;

export const NewsContentFileContainer = styled.div`
  width: 100%;
  margin-bottom: 32px;
  @media (max-width: ${BREAKPOINTS.IPAD_LANDSCAPE}) {
    padding: 0 5px;
  }
`;

export const NewsContentFileHeader = styled.div`
  font-size: 16px;
  font-weight: 700;
  color: ${COLORS.GRAY_2};
  margin-bottom: 16px;
`;

export const NewsContentFileList = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  padding: 10px 15px;
  background-color: #f5f5f5;
  margin-bottom: 8px;
`;

export const NewsContentFileNameContainer = styled.div`
  display: flex;
  align-items: center;
`;

export const NewsContentFileName = styled.div`
  font-size: 14px;
  font-weight: 700;
  color: ${COLORS.GRAY_1};
`;

export const OtherNewsContainer = styled.div`
  width: 100%;
  margin-bottom: 36px;
`;

export const OtherNewsHeaderContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 36px;
  @media (max-width: ${BREAKPOINTS.IPAD_LANDSCAPE}) {
    padding: 0 25px;
    flex-direction: column;
  }
`;

export const OtherNewsHeader = styled.div`
  font-size: 22px;
  font-weight: 700;
  color: ${COLORS.GRAY_2};
`;

export const DynamicCotentZoneContainer = styled.div`
  width: 100%;
`;
