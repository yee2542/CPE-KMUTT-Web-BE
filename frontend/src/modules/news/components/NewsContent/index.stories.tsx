import { Meta, Story } from '@storybook/react/types-6-0';

import { GlobalStyle } from 'modules/root/pages/App/styled';

import { NewsContentProps } from './types';

import NewsContent from '.';

export default {
  title: 'Modules/News/Components/NewsContent',
  component: NewsContent,
} as Meta;

export const NewsContentStory: Story<NewsContentProps> = () => (
  <>
    <GlobalStyle />
    <NewsContent tumbnail="/images/thumbnail2.png" />
  </>
);
