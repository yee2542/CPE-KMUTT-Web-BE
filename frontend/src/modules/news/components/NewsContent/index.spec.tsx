import React from 'react';

import { render, shallow } from 'enzyme';

import Container from 'common/components/Container';

import {
  NewsContentTumbnail,
  NewsContentContainer,
  NewsContentFileContainer,
  NewsContentFileHeader,
  NewsContentFileList,
  NewsContentFileName,
  NewsContentFileNameContainer,
  NewsContentHeader,
  NewsContentPostDate,
  NewsContentStyle,
} from './styled';

import NewsContent from '.';

describe('NewsContent Compnent', () => {
  it('NewsContent Compnent should be defined', () => {
    const wrap = shallow(<NewsContent tumbnail="/images/thumbnail2.png" />);
    expect(wrap.exists()).toBe(true);
  });

  it('NewsContent shoyld matched snapshots', () => {
    const wrap = render(<NewsContent tumbnail="/images/thumbnail2.png" />);
    expect(wrap).toMatchSnapshot();
  });

  it('NewsContent Compnent should import these components', () => {
    const wrap = shallow(<NewsContent tumbnail="/images/thumbnail2.png" />);
    expect(wrap.find(NewsContentTumbnail).exists()).toBe(true);
    expect(wrap.find(Container).exists()).toBe(true);
    // expect(wrap.find(NewsContentContainer).exists()).toBe(true);
    // expect(wrap.find(NewsContentFileContainer).exists()).toBe(true);
    // expect(wrap.find(NewsContentFileHeader).exists()).toBe(true);
    // expect(wrap.find(NewsContentFileList).exists()).toBe(true);
    // expect(wrap.find(NewsContentFileName).exists()).toBe(true);
    // expect(wrap.find(NewsContentFileNameContainer).exists()).toBe(true);
    expect(wrap.find(NewsContentHeader).exists()).toBe(true);
    expect(wrap.find(NewsContentPostDate).exists()).toBe(true);
    expect(wrap.find(NewsContentStyle).exists()).toBe(true);
  });

  it('NewsContent should show content section when have data from API', () => {
    const wrap = shallow(
      <NewsContent
        tumbnail="/images/thumbnail2.png"
        contents={[<NewsContentContainer key={'1'}>news content</NewsContentContainer>]}
      />,
    );
    expect(wrap.find(NewsContentContainer).exists()).toBe(true);
  });

  it('NewsContent should show download section when have data from API', () => {
    const wrap = shallow(<NewsContent file={[{ title: 'test', link: '/' }]} />);
    expect(wrap.find(NewsContentFileContainer).exists()).toBe(true);
    expect(wrap.find(NewsContentFileHeader).exists()).toBe(true);
    expect(wrap.find(NewsContentFileList).exists()).toBe(true);
    expect(wrap.find(NewsContentFileName).exists()).toBe(true);
    expect(wrap.find(NewsContentFileNameContainer).exists()).toBe(true);
  });
});
