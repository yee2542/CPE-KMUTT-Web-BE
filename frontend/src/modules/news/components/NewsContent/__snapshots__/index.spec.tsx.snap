// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContent Compnent NewsContent shoyld matched snapshots 1`] = `
<div
  class="styled__NewsContentStyle-sc-18h3naq-0 gkPTfa"
>
  <div
    class="styled__ContainerStyled-sc-44j5kv-0 gjcMjy"
  >
    <h1
      class="styled__NewsContentHeader-sc-18h3naq-1 iXHTZA"
    />
    <div
      class="styled__NewsContentPostDate-sc-18h3naq-2 bozAvN"
    />
    <div
      class="styled__NewsContentTumbnail-sc-18h3naq-3 daQSRC"
      style="background-image:url(/images/thumbnail2.png)"
    />
  </div>
</div>
`;
